### PR TITLE
Fe signup login

### PR DESCRIPTION
**Pull request by williamlee21**
Back end
- Add user model in app.py
- Add login and signup routes in app.py
- Returns token and validations errors

Front end
- Stores token to local storage on login/signup success

UI
- Show warning if user enter invalid email or password
- Hides user password input

Improvement
- Separate user model from app.py

### CODE REVIEW

File: server/app.py [hunk 1]
```diff
@@ -1,13 +1,85 @@
...
 
 app = Flask(__name__)
 app.config['SQLALCHEMY_DATABASE_URI'] = 'postgresql://{}:{}@localhost/{}'.format(DB_USERNAME, DB_PASSWORD, DB_NAME)
 app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
-db.init_app(app)
+app.config['SECRET_KEY'] = 'bubblegum'
+db= SQLAlchemy(app)
+bcrypt = Bcrypt(app)
+
+class User(db.Model):
```
Comment (isabelt220) [inline server/app.py]: I think you should put the User model in models.py and also I think there should be a reference to the receipts the user has created.

File: server/app.py [hunk 2]
```diff
@@ -1,13 +1,85 @@
...
+  body = json.loads(request.get_data())
+  EMAIL_REGEX = re.compile(r'^[a-zA-Z0-9.+_-]+@[a-zA-Z0-9._-]+\.[a-zA-Z]+$')
+  print(EMAIL_REGEX.match(body['email']), 'here')
+  if EMAIL_REGEX.match(body['email']) == None: 
+    return jsonify({'err':'Please enter a valid email'})
+  if len(body["password"]) < 6:
+    return jsonify({'err': 'Password must be 6 letters or more'})
+  user = User(body["email"], body["password"])
+  db.session.add(user)
+  db.session.commit()
```
Comment (isabelt220) [inline server/app.py]: I think you should have a statement that checks whether the signup was successful or not. The user information might not be committed to the database successfully.

File: server/app.py [hunk 3]
```diff
@@ -1,13 +1,85 @@
...
+
+  token = jwt.encode(body, app.config['SECRET_KEY']).decode("utf-8")
+  print(token, type(token))
+  return jsonify({'token': token})
+  
+@app.route('/login', methods=['GET', 'POST'])
```
Comment (isabelt220) [inline server/app.py]: what does the GET request do?

File: server/app.py (hunk 1 above)
Comment (williamlee21) [reply server/app.py]: I was having an issue trying to separate the model from the app.py and will look into it.
The user model should have a reference to the receipts model?

File: server/app.py (hunk 3 above)
Comment (williamlee21) [reply server/app.py]: I don't think it does anything. I'll delete that.

File: server/app.py (hunk 2 above)
Comment (williamlee21) [reply server/app.py]: Okay.

**Comment by ethanlow23** [comment]
I merged the updated structure of the UI so that the dashboard and other components after user logs in does not reside in the app component. It should be reflected in the dev branch

File: client/src/components/Snackbar.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    if (reason === 'clickaway') {
+      return;
+    }
+
+    setOpen(false);
+  };
+
+  return (
+    <div>
+      {/* <Button
```
Comment (bonnieli) [inline client/src/components/Snackbar.js:111]: some commented out code

**Review by bonnieli — APPROVED** [review]

**Pull request merged**